pkg/captcha/captcha.go

```diff
@@ -0,0 +1,165 @@
+// Package captcha implements utility for captcha management.
+package captcha
+
+import (
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"net/http"
+	"net/url"
+	"os"
+
+	cache "github.com/maxlerebourg/crowdsec-bouncer-traefik-plugin/pkg/cache"
+	configuration "github.com/maxlerebourg/crowdsec-bouncer-traefik-plugin/pkg/configuration"
+	logger "github.com/maxlerebourg/crowdsec-bouncer-traefik-plugin/pkg/logger"
+)
+
+// Client Captcha client.
+type Client struct {
+	Valid              bool
+	provider           string
+	siteKey            string
+	secretKey          string
+	gracePeriodSeconds int64
+	htmlPage           *template.Template
+	cacheClient        *cache.Client
+	httpClient         *http.Client
+	log                *logger.Log
+}
+
+type infoProvider struct {
+	js       string
+	key      string
+	validate string
+}
+
+var (
+	//nolint:gochecknoglobals
+	captcha = map[string]infoProvider{
+		configuration.HcaptchaProvider: {
+			js:       "https://hcaptcha.com/1/api.js",
+			key:      "h-captcha",
+			validate: "https://api.hcaptcha.com/siteverify",
+		},
+		configuration.RecaptchaProvider: {
+			js:       "https://www.google.com/recaptcha/api.js",
+			key:      "g-recaptcha",
+			validate: "https://www.google.com/recaptcha/api/siteverify",
+		},
+		configuration.TurnstileProvider: {
+			js:       "https://challenges.cloudflare.com/turnstile/v0/api.js",
+			key:      "cf-captcha",
+			validate: "https://challenges.cloudflare.com/turnstile/v0/siteverify",
+		},
+	}
+)
+
+func compileTemplate(path string) (*template.Template, error) {
+	var err error
+	if path == "" {
+		return nil, fmt.Errorf("no captcha template provided")
+	}
+	//nolint:gosec
```

Comment:
i'll look at this, //nolint:gosec

Reply:
The main security issue found here is if the path is not checked it could access other elements on the system.  
However the path is accessed at configuration time. So if an attacker has access to configuration and can start/stop traefik, something wrong happened before